onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -266,16 +535,24 @@ Status Transpose::Compute(OpKernelContext* ctx) const {
   std::vector<int64_t> output_dims(rank);
   const std::vector<size_t>* p_perm;
   std::vector<size_t> default_perm(rank);
-  const auto& status = ComputeOutputShape(X, output_dims, default_perm, p_perm);
+  Status status = ComputeOutputShape(X, output_dims, default_perm, p_perm);
   if (!status.IsOK())
     return status;
 
   TensorShape output_shape{output_dims};
   Tensor& Y = *ctx->Output(0, output_shape);
 
-  DoUntypedTranspose(*p_perm, X, Y);
+  size_t from = 0, to = 0;
+  bool moving_single_axis = IsMovingSingleAxis(*p_perm, from, to);
 
-  return Status::OK();
+  if (moving_single_axis && X.GetElementType() != utils::GetONNXTensorElementDataType<std::string>()) {
```

Comment:
>GetONNXTensorElementDataType [](start = 57, length = 28)

The enums populated are actually from TensorProto_DataType. This may or may not match,

Reply:
```
// Copied from TensorProto::DataType
// Currently, Ort doesn't support complex64, complex128, bfloat16 types
typedef enum ONNXTensorElementDataType {
```

Seems like they should always be in sync and instead of having both utils::ToTensorProtoElementType and utils::GetONNXTensorElementDataType we could just have one helper as the current setup is confusing.